### PR TITLE
customResponseIncomingCallMap

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Add rpc.js.flow to gitattributes to mark as generated
- [x] Split calls we want to handle responses to. Previously we used incomingCallMap and you could pass a function of the form `(params, response, state)` or `(params, void, state)` which made the typing pretty weak.  Now we have incomingCallMap which assumes you aren't handling the response and customResponseIncomingCallMap which does give you the response object
- [x] Additionally removed state from being passed down. We return effects from all these things so if you need to do something complicated and need state you can get it yourself by returning a Saga.call(). I think ultimately it might be better for these callback handlers to be much smaller and just dispatch a small action , then the saga handler can just do what it needs. This allows us to remove TypedState from the generated code entirely (it was a type parameter before which engine passed but nothing else did, effectively rendering it `empty` as far as flow was concerned which masks some real issues.

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by zanderz** [comment]
LGTM. I wonder if there's a way to mark more of the existing code that accesses the store as being a deprecated technique, it's hard to know sometimes which stuff to copy.

File: shared/engine/session.js [hunk 1]
```diff
@@ -158,7 +161,8 @@ class Session {
   }
 
   // We have an incoming call tied to a sessionID, called only by engine
-  incomingCall(method: MethodKey, param: Object, response: ?Object): boolean {
+  incomingCall(method: MethodKey, param: Object, _response: ?Object): boolean {
+    let response = _response
```
Comment (akalin-keybase) [inline shared/engine/session.js]: why have both `_response` and `response`?

Comment (chrisnojima) [reply shared/engine/session.js]: oh thats leftover, i was nulling it out under some circumstances and we have flow assume parameters are const so i had to have another var, but thats been removed so i'll just remove it, tnx!

**Comment by chrisnojima** [comment]
@akalin @zanderz note a fix in the last commit (https://github.com/keybase/client/pull/13742/commits/7909472084472bd84697afe3c26f7de0a76f53d2). I moved the coreHandlers thing out of engine and into config (which is basically app wide stuff) and now call response.result() in the incoming handler case (its handled in the rpcsaga stuff already). This fixes logui not getting a response

**Pull request merged**